grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -328,52 +328,75 @@ class QueryService implements Status {
         results as Collection<ConsentCollectionLink>
     }
 
+    Collection<ConsentCollectionLink> findCollectionLinkById(String consentCollectionLinkId) {
+        final session = sessionFactory.currentSession
+        final String query =
+                ' select * from consent_collection_link c ' +
+                        ' where c.id = :consentCollectionLinkId '
+        final sqlQuery = session.createSQLQuery(query)
+        final results = sqlQuery.with {
+            addEntity(ConsentCollectionLink)
+            setString('consentCollectionLinkId', consentCollectionLinkId)
+            list()
+        }
+        results as Collection<ConsentCollectionLink>
```

Comment:
In this case, you know that you're only ever going to get a single result. I think it would be better to get the first element of the list, if not empty, and return that. The name of this method implies that you will be getting only a single `ConsentCollectionLink`

Reply:
Oh! I saw this late, I'll change this in my next PR Greg.